crates/core/src/operations/mod.rs

```diff
@@ -62,10 +66,61 @@ pub mod update;
 pub mod write;
 pub mod writer;
 
+#[async_trait]
+pub trait CustomExecuteHandler: Send + Sync {
+    // Execute arbitrary code at the start of a delta operation
+    async fn pre_execute(&self, log_store: &LogStoreRef, operation_id: Uuid) -> DeltaResult<()>;
+
+    // Execute arbitrary code at the end of a delta operation
+    async fn post_execute(&self, log_store: &LogStoreRef, operation_id: Uuid) -> DeltaResult<()>;
+
+    // Execute arbitrary code at the start of the post commit hook
+    async fn before_post_commit_hook(
+        &self,
+        log_store: &LogStoreRef,
+        file_operation: bool,
+        operation_id: Uuid,
+    ) -> DeltaResult<()>;
+
+    // Execute arbitrary code at the end of the post commit hook
+    async fn after_post_commit_hook(
+        &self,
+        log_store: &LogStoreRef,
+        file_operation: bool,
+        operation_id: Uuid,
+    ) -> DeltaResult<()>;
+}
+
 #[allow(unused)]
 /// The [Operation] trait defines common behaviors that all operations builders
 /// should have consistent
-pub(crate) trait Operation<State>: std::future::IntoFuture {}
+pub(crate) trait Operation<State>: std::future::IntoFuture {
+    fn get_log_store(&self) -> &LogStoreRef;
```

Comment:
I'm going to make a suggestion and I realize that it will require a lot of copypasta :laughing: 

```suggestion
    fn log_store(&self) -> LogStoreRef;
```

This would be symmetrical with the function on the DeltaTable itself :smile: 

I would recommend that the implementers also return self.log_store.clone() since `LogStoreRef` is an`Arc<dyn LogStore>`